account_invoice_product_supplier_price_update/account_invoice.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+# © 2016 Chafique DELLI @ Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api, _
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    @api.model
+    def _check_product_supplierprice(self):
+        lines = []
+        for line in self.invoice_line:
+            current_price_unit = []
+            suppinfo = self.env['product.supplierinfo'].browse(False)
+            supplier = self.partner_id.commercial_partner_id or self.partner_id
+            for seller in line.product_id.seller_ids:
```

Comment:
something like that should more rapid:

    for seller in line.product_id.seller_ids.filtered(lambda seller: seller.name== supplier):